### PR TITLE
Separate aks and fleet in containerservice

### PR DESCRIPTION
**Pull request by FumingZhang**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, reformat folder structure under namespace `Microsoft.Containerservice` to separate aks and fleet into two parts with separate Autorest configs.
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  No new service/feature.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  ASAP.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.go.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+``` yaml $(go) && !$(track2)
+go:
+  license-header: MICROSOFT_MIT_NO_VERSION
+  clear-output-folder: true
+```
+
+``` yaml $(go) && $(track2)
+license-header: MICROSOFT_MIT_NO_VERSION
+module-name: sdk/resourcemanager/containerservice/armcontainerservicefleet
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.go.md:13]: New `readme.go` for fleet, modified `module-name` from `sdk/resourcemanager/containerservice/armcontainerservice` to `sdk/resourcemanager/containerservice/armcontainerservicefleet`.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.go.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+module: github.com/Azure/azure-sdk-for-go/$(module-name)
+output-folder: $(go-sdk-folder)/$(module-name)
+azure-arm: true
+```
+
+### Go multi-api
+
+``` yaml $(go) && $(multiapi)
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.go.md:21]: Keeping tags previously released by track 1 SDK.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.go.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+### Tag: package-preview-2022-07 and go
+
+These settings apply only when `--package-preview-2022-07 --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag)=='package-preview-2022-07' && $(go)
+namespace: containerservice
+output-folder: $(go-sdk-folder)/services/preview/$(namespace)/mgmt/2022-07-02-preview/$(namespace)fleet
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.go.md:34]: Append `fleet` to the end of `output-folder`.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.java.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
+## Java
+
+These settings apply only when `--java` is specified on the command line.
+Please also specify `--azure-libraries-for-java-folder=<path to the root directory of your azure-libraries-for-java clone>`.
+
+``` yaml $(java)
+azure-arm: true
+fluent: true
+namespace: com.microsoft.azure.management.containerservicefleet
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.java.md:9]: New `readme.java` for fleet, modified 
- `namespace` from `com.microsoft.azure.management.containerservice` to `com.microsoft.azure.management.containerservicefleet`
- `output-folder` from `$(azure-libraries-for-java-folder)/azure-mgmt-containerservice` to `$(azure-libraries-for-java-folder)/azure-mgmt-containerservicefleet`
- `title` from `ContainerServiceManagementClient` to `ContainerServiceFleetManagementClient`

Seems not able to keep namespace `containerservice` like golang.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.java.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+### Tag: package-preview-2022-09 and java
+
+These settings apply only when `--tag=package-preview-2022-09` is specified on the command line.
+Please also specify `--azure-libraries-for-java-folder=<path to the root directory of your azure-sdk-for-java clone>`.
+
+``` yaml $(tag) == 'package-preview-2022-09' && $(java) && $(multiapi)
+java:
+  namespace: com.microsoft.azure.management.containerservicefleet.v2022_09_02_preview
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.java.md:33]: Append `fleet` to namespace and output-folder.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
+# ContainerServices
+
+> see https://aka.ms/autorest
+
+This is the AutoRest configuration file for ContainerServices-Fleet.
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.md:5]: Update global description.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.nodejs.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+## Node.js
+
+These settings apply only when `--nodejs` is specified on the command line.
+Please also specify `--node-sdks-folder=<path to root folder of your azure-sdk-for-node clone>`.
+
+``` yaml $(nodejs)
+nodejs:
+  azure-arm: true
+  package-name: azure-arm-containerservicefleet
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.nodejs.md:9]: New `readme.nodejs` for fleet, is this file actually used? Modified `package-name` and `output-folder`.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.python.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
+## Python
+
+These settings apply only when `--python` is specified on the command line.
+
+```yaml $(python)
+azure-arm: true
+license-header: MICROSOFT_MIT_NO_VERSION
+package-name: azure-mgmt-containerservicefleet
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.python.md:8]: New `readme.python` for fleet, modified `package-name` from `azure-mgmt-containerservice` to `azure-mgmt-containerservicefleet`.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.python.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+perform-load: false
+```
+
+### Tag: package-preview-2022-09 and python
+
+These settings apply only when `--tag=package-preview-2022-09 --python` is specified on the command line.
+Please also specify `--python-sdks-folder=<path to the root directory of your azure-sdk-for-python clone>`.
+
+``` yaml $(tag) == 'package-preview-2022-09' && $(python)
+namespace: azure.mgmt.containerservicefleet.v2022_09_02_preview
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.python.md:38]: Append fleet to `namespace` and `output-folder`.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.python.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+multiapi: true
+batch:
+  - tag: package-preview-2022-09
+  - tag: package-preview-2022-07
+  - tag: package-preview-2022-06
+  - multiapiscript: true
+```
+
+``` yaml $(multiapiscript)
+output-folder: $(python-sdks-folder)/containerservice/azure-mgmt-containerservicefleet/azure/mgmt/containerservicefleet/
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.python.md:28]: Append fleet to `output-folder`.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.typescript.md [hunk 11]
```diff
@@ -0,0 +1,12 @@
+## TypeScript
+
+These settings apply only when `--typescript` is specified on the command line.
+Please also specify `--typescript-sdks-folder=<path to root folder of your azure-sdk-for-js clone>`.
+
+``` yaml $(typescript)
+typescript:
+  azure-arm: true
+  package-name: "@azure/arm-containerservicefleet"
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/readme.typescript.md:9]: New readme.typescript for fleet, modified `package-name` and `output-folder` to append fleet.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/readme.md [hunk 12]
```diff
@@ -2,14 +2,12 @@
 
 > see https://aka.ms/autorest
 
-This is the AutoRest configuration file for ContainerServices (ACS/AKS/OpenShift).
+This is the AutoRest configuration file for ContainerServices-AKS (ACS and OpenShift are deprecated).
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/readme.md:5]: Update global description.

**Comment by FumingZhang** [comment]
In this PR, the 2 services (aks, fleet) under namespace Microsoft.ContainerService are split into 2 separate subfolders. Add new Autorest configuration files (readmes) for fleet to generate independent SDKs. No changes to the content of existing swagger files.

**Comment by FumingZhang** [comment]
For the 12 failed checks,

- [Breaking Change(Cross-Version)](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088176968), list items are all as expected, they were reviewed when creating the swagger for these API versions. Also no change to the content of existing swagger files in this PR.
- [PR Summary](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088178975), it failed due to `no such file or directory, open 'specification/containerservice/resource-manager/readme.md'`, which is exactly what this PR changed (the readmes are moved to subfolders), so is this error tolerable?
- [SDK azure-powershell](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088228379), similar error as occurred in previous swagger releases (e.g., add API version [2022-09-01](https://github.com/Azure/azure-rest-api-specs/pull/21148)). I did not (and did not before) configure powershell to generate corresponding products for each of the two services.
- [SDK azure-resource-manager-schemas](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088227397), error msg is `cannot force update branch 'sdkAuto/21259/containerservice' as it is the current HEAD of the repository.` seems like a temporary error to me
- [SDK azure-sdk-for-net ](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088226970), similar error occurred in previous swagger releases (e.g., add API [2022-09-01](https://github.com/Azure/azure-rest-api-specs/pull/21148)). I suppose this check is for track 1 SDK, and .NET has released track 2 SDK for [containerservice](https://www.nuget.org/packages/Azure.ResourceManager.ContainerService/1.0.0).
- [Swagger Avocado](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088177347), all 100+ items are about [MISSING_README](https://github.com/Azure/avocado/blob/master/README.md#MISSING_README). Do I put the readmes in the wrong folder? But it seems SDK could be generated successfully.
- [Swagger BreakingChange](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088176875), 3637 errors, look like caused by moving the swagger files, since no change to the content of existing swagger files in this PR.
- [Swagger LintDiff](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088177163), 3621 errors, due to the change of the file path, the places in the swagger files corresponding to the old API versions that do not conform to the current specification are scanned out. No change to the content of existing swagger files in this PR.
- [Swagger ModelValidation](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088177693), 1556 errors, similar to the previous two checks. No change to the content of existing swagger files in this PR.
- [Swagger SDK Breaking Change Tracking](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088178669), unused model definitions are removed in go SDK, which is somehow expected (also happened in add API version [2022-09-01](https://github.com/Azure/azure-rest-api-specs/pull/21148)). Another issue is fleet related operations are removed from Python SDK azure-mgmt-containerservice, which is also expected, since they are moving to a new SDK named azure-mgmt-containerservicefleet.
- [Swagger SpellCheck](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088178506), the word `Ochestrator` (used in 2016-03-30 API) needs to be added to custom_words.txt.
- [~[Staging] Swagger LintDiff](https://github.com/Azure/azure-rest-api-specs/pull/21259/checks?check_run_id=9088177253), 6814 errors, there is something wrong with the check itself, some normal annotations are also marked as errors (e.g., 'arm-id' is not considered as a valid format).

**Comment by akning-ms** [comment]
> In this PR, the 2 services (aks, fleet) under namespace Microsoft.ContainerService are split into 2 separate subfolders. Add new Autorest configuration files (readmes) for fleet to generate independent SDKs. No changes to the content of existing swagger files.

As seems SDK has been split into 2 packages. + different SDK lang owner to confirm:  @weidongxu-microsoft @msyyc @qiaozha @tadelesh @fengzhou-msft to confirm and review readme

**Comment by weidongxu-microsoft** [comment]
Guide to ask for approval of the new namespace for SDK https://azure.github.io/azure-sdk/policies_reviewprocess.html

**Pull request closed without merging**